onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -8,12 +8,59 @@
 #include "core/common/optional.h"
 #include "core/framework/op_kernel.h"
 #include "core/providers/cpu/containers.h"
+#include "core/util/math.h"
 #include "core/util/math_cpuonly.h"
 #include "core/platform/threadpool.h"
+#include "core/common/safeint.h"
 #include <cmath>
 
 namespace onnxruntime {
 
+/* Common verficiations. */
+void OrtEnforce_ReduceAggregatorKR(const std::vector<int64_t>& fast_shape, const Tensor& output);
+void OrtEnforce_ReduceAggregatorRK(const std::vector<int64_t>& fast_shape, const Tensor& output);
```

Comment:
nit: something like ValidateReduceAggregatorKRInput would conform to the coding conventions. we allow '_' in unit test names but avoid it in the production code.

Reply:
done